numpy/core/tests/test_regression.py

```diff
@@ -2500,3 +2500,12 @@ def __len__(self):
         t = T()
         #gh-13659, would raise in broadcasting [x=t for x in result]
         np.array([t])
+
+    @pytest.mark.skipif(sys.maxsize < 2 ** 31 + 1, reason='overflows 32-bit python')
+    @pytest.mark.skipif(sys.platform == 'win32', reason='overflows on windows')
```

Comment:
Is this second check necessary? It doesn't make sense to me why this would overflow.

Reply:
Looks like it was fixed in https://github.com/python/cpython/pull/3006. Can we combine this condition with `sys.version_info[:1] < 3.8` and add a reference to bpo-16865?